gwpy/timeseries/timeseries.py

```diff
@@ -951,6 +952,32 @@ def resample(self, rate, window='hamming', ftype='fir', n=None):
             new.sample_rate = rate
             return new
 
+    def upsample_repeat(self, factor):
+        """Upsample this TimeSeries by repeating elements by an integer factor.
+
+        This will return a new TimeSeries with a new sample_rate equal to the
+        previous one multiplied by factor.
+
+        Parameters
+        ----------
+        factor: `int`
+                The number of times to repeat each element
+
+        Returns
+        -------
+        TimeSeries
+            a new Series with the resampling applied, and the same
+            metadata
+        """
+        if not isinstance(factor, numbers.Integral):
+            raise ValueError("Cannot upsample_repeat with non-integer factor.")
+        repeat_inds = numpy.repeat(numpy.arange(len(self)), factor)
+        new = self[repeat_inds]  # xindex should also be handled here
+        new.__metadata_finalize__(self)
+        new._unit = self.unit
```

Comment:
I hope this shouldn't be required, can you test without it. If it _is_ required, maybe that indicates a hole in the `__getitem__` overrides for `Series`.